linux/utils/device_connectable_enable.yml

```diff
@@ -12,20 +12,45 @@
         value: "FALSE"
 
 # Check if above advanced settings already configured
-- include_tasks: ../../common/vm_get_extra_config.yml
+- name: "Get VM's extra config"
+  include_tasks: ../../common/vm_get_extra_config.yml
+
 - name: "Set fact of existence of expected extra config"
   ansible.builtin.set_fact:
     vm_advanced_settings_exist: true
   when:
     - vm_advanced_settings[0].key in vm_extra_config
-    - vm_extra_config[vm_advanced_settings[0].key] | lower == vm_advanced_settings[0].value | lower
+    - vm_extra_config[vm_advanced_settings[0].key] == vm_advanced_settings[0].value
     - vm_advanced_settings[1].key in vm_extra_config
-    - vm_extra_config[vm_advanced_settings[1].key] | lower == vm_advanced_settings[1].value | lower
+    - vm_extra_config[vm_advanced_settings[1].key] == vm_advanced_settings[1].value
 
 # Set key-value pairs when not in VMX file
-- block:
-    - include_tasks: shutdown.yml
-    - include_tasks: ../../common/vm_set_extra_config.yml
-    - include_tasks: ../../common/vm_wait_guest_ip.yml
-    - include_tasks: ../../common/update_inventory.yml
+- name: "Add isolation.device settings to VM's extra config"
+  block:
+    - name: "Shutdown OS"
+      include_tasks: shutdown.yml
+
+    - name: "Add isolation.device settings to VM's extra config"
+      include_tasks: ../../common/vm_set_extra_config.yml
+
+    - name: "Power on VM"
+      include_tasks: ../../common/vm_set_power_state.yml
+      vars:
+        vm_power_state_set: 'powered-on'
+
+    - name: "Get VM's extra config"
+      include_tasks: ../../common/vm_get_extra_config.yml
+
+    - name: "Check isolation.device settings are present in VM's extra config after VM power-on"
+      ansible.builtin.assert:
+        that:
+          - vm_extra_config['isolation.device.connectable.disable'] is defined
+          - vm_extra_config['isolation.device.connectable.disable'] == 'FALSE'
+          - vm_extra_config['isolation.device.edit.disable'] is defined
+          - vm_extra_config['isolation.device.edit.disable'] == 'FALSE'
+        fail_msg: >-
+          The isolation.device settings are not correct in VM's extra config after VM power-on.
+
+    - name: "Update VM's IP in inventory"
```

Comment:
I think this task can be put after power on to check VM booting up normally and successfully, then check config.

Reply:
Moved up task.